### PR TITLE
Remove unused function

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2616" title="DX-2616" target="_blank">BugDX-2616</a>  Buildscript conflict resolution appears to get trumped by API
  </td></table>
  <br />
 

Looks like this was inadvertantly fixed in https://github.com/ActiveState/cli/pull/3249. I've updated the PR to just remove the unused code discovered during PR review.

### CODE REVIEW

File: internal/runners/pull/pull.go [hunk 1]
```diff
@@ -272,7 +272,18 @@ func (p *Pull) mergeBuildScript(remoteCommit, localCommit strfmt.UUID) error {
...
+	err = buildscript.Update(p.project, atTime, mergedExpr, p.auth)
+	if err != nil {
+		return "", errs.Wrap(err, "Could not update build script with merged changes")
+	}
+
+	if synced, err := buildscriptRunbits.Sync(p.project, nil, p.out, p.auth); err == nil && synced {
+		return localcommit.Get(p.project.Dir())
+	} else {
+		return "", errs.Wrap(err, "Could not commit merged build script")
+	}
```
Comment (mitchell-as) [inline internal/runners/pull/pull.go:284]: This change here may cause this ticket to be blocked on https://activestatef.atlassian.net/browse/DX-2619, where we want to do away with `buildscript.Sync()`.

File: version.txt [hunk 2]
```diff
@@ -1 +1 @@
-0.45.0-RC1
\ No newline at end of file
+0.45.0-RC1
```
Comment (mitchell-as) [inline version.txt:1]: Needed to satisfy the "Check Format" action...

**Review by Naatan — CHANGES_REQUESTED** [review]
In addition to comments, as noted on Slack; please make the buildscript being dirty an error condition for `state pull`, do not try to resolve conflicts with an uncommitted buildscript.

**Comment by mitchell-as** [comment]
After addressing PR review comments and merging remote changes, it looks like this whole thing was inadvertantly fixed in https://github.com/ActiveState/cli/pull/3249. I've updated the PR to just remove the unused code discovered during PR review.

**Review by Naatan — APPROVED** [review]

**Comment by mitchell-as** [comment]
Please re-approve. I had to merge in v45's buildplan refactor as there was a conflict in this file related to file imports. I've resolve them here.

**Review by Naatan — APPROVED** [review]

**Pull request merged**